File: problem_4.rb
def palindrome? (number)
  number.to_s == (number.to_s).reverse
end

upper_limit = 1000
a = []
(1...upper_limit).each do |x|
  (1...upper_limit).each do |y|
    product = (x * y)
    a << product if palindrome?(product)
  end
end

puts a.last

a = (1...upper_limit).map do |x|
  (1...upper_limit).map do |y|
    product = (x * y)
    product if palindrome?(product)
  end
end.flatten.reject(&:nil?)

puts a.last

File: problem_4.clj
(defn palindrome? [number]
  (= (reverse (str number)) (seq (str number))
  )
)

(palindrome? 1111)

(let [upper-limit 1000]
  (last (for [x (range 1 upper-limit) y (range 1 upper-limit) :when (palindrome?(* x y))] (* x y)))
)
